src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -970,9 +971,29 @@ public String getName() {
         @Override
         public GroupDetailsImpl mapAttributes(String dn, Attributes attributes) throws NamingException {
             LdapName name = new LdapName(dn);
-            String groupName = fixGroupname(String.valueOf(name.getRdn(name.size() - 1).getValue()));
+            String groupName = fixGroupname(extractGroupName(name, attributes));
             return new GroupDetailsImpl(dn, groupName);
         }
+
+        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            boolean isCN = false;
+            String groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            Attribute cnAttribute = attributes.get(CN);
+            if (cnAttribute != null) {
+                for (NamingEnumeration e = cnAttribute.getAll(); e.hasMore() && !isCN;) {
```

Comment:
This still looks unnecessarily complex to me.

Reply:
Please, let me know what you have in mind to simply it.